### ISSUE TITLE
Issues started appearing with subscriptions

### ISSUE THREAD

**Issue by olivoil** [aws-amplify/amplify-cli#4604]
**Describe the bug**
Subscriptions used to work on our iOS application, but very recently have started sending errors. We did not push any code updates to amplify or the iOS app, and had tested these subscriptions after our last deployments, at which point they were working well.  

**Amplify CLI Version**
4.21.3

**To Reproduce**

**Expected behavior**

**Screenshots**
Log output of what is being received in subscriptions events from iOS:

```
▿ AppSyncResponse
 ▿ id : Optional<String>
  - some : "8F6DCE9A-A6F3-45AF-823E-7875DCF81C7F"
 ▿ payload : Optional<Dictionary<String, AppSyncJSONValue>>
  ▿ some : 1 element
   ▿ 0 : 2 elements
    - key : "errors"
    ▿ value : AppSyncJSONValue
     ▿ array : 1 element
      ▿ 0 : AppSyncJSONValue
       - string : "Found 2 problem(s) with the schema:\nThere is no type resolver defined for interface / union \'PlanEvent\' type.\nThere is no type resolver defined for interface / union \'CommunityEvent\' type."
 - responseType : AppSyncRealTimeClient.AppSyncResponseType.data
```

These 2 types `PlanEvent` and `CommunityEvent` are unions of @model types defined in the schema. They are not directly related to what the subscriptions send. 

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version: v12.18.0

**Additional context**

**Comment by kcwinner** [comment]
@olivoil  I had issues the other day. I also posted this in the discord channel
```
KenWin0x53906/16/2020
Ok so I have solved the underlying issue. It appears to be a bad schema issue using an interface - there is no type resolver defined for interface
...
```

Schema had been working since May 18th. My only guess is something in the underlying AppSync service changed and now what used to not throw errors, does.

Above you posted the AppSyncResponse with:
```
"Found 2 problem(s) with the schema:\nThere is no type resolver defined for interface / union \'PlanEvent\' type.\nThere is no type resolver defined for interface / union \'CommunityEvent\' type."
 - responseType : AppSyncRealTimeClient.AppSyncResponseType.data
```

If you fix the schema errors/issues laid out in the response, subscriptions should begin working again. Without knowing your schema I'm not sure what it needs to look like but the error is complaining about the type resolvers for `PlanEvent` and `CommunityEvent`

I ended up simply removing mine. Perhaps, though, it is a bug?
A larger concern here is that nowhere else have I seen the error pop up. The schema doesn't complain and it doesn't complain on queries or mutations. Only on subscriptions - and it breaks all subscriptions.

Here was my walkthrough of the issue:
https://twitter.com/KenWin0x539/status/1272970527347523587?s=20
